x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -837,7 +871,21 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
           );
           if (pkgInfo) {
             validatePackagePolicyOrThrow(packagePolicy, pkgInfo);
-
+            const { secretsStorage: secretsStorageEnabled } =
+              appContextService.getExperimentalFeatures();
+            if (secretsStorageEnabled) {
+              const secretsRes = await extractAndUpdateSecrets({
```

Comment:
could we add an integration test for the `bulkUpdate` as well?

Reply:
It isn't exposed via any APIs, it's only called by the synthetics service [here](https://github.com/hop-dev/kibana/blob/f59471bcdc8d8fa156e7997749efe7d252457d77/x-pack/plugins/synthetics/server/synthetics_service/private_location/synthetics_private_location.ts#L332)